parlai/torchscript/modules.py

```diff
@@ -750,6 +755,188 @@ def utf8_chars(self, s: str) -> List[str]:
         return chars
 
 
+@torch.jit.script
+class ScriptableSubwordBpeHelper(object):
+    """
+    Version of parlai.utils.bpe.ScriptableSubwordBpeHelper that can be TorchScripted.
+    """
+    @classmethod
+    def findall(cls, text: str) -> List[str]:
+        """
+        Split tokens in a manner that replicates parlai.utils.bpe.Gpt2BpeHelper.
+        """
+        tokens: List[str] = []
+        idx = 0
+        num_passes = 0
+        while idx < len(text):
+            num_passes += 1
+            if num_passes > 10000:
+                return ["*** Infinite loop in ScriptableSubwordBpeHelper.findall()! ***"]
+            if not text[idx].isspace():
+                # Capture runs of one type of character
+                if text[idx] == " ":
+                    last_matching_idx = idx + 1
+                else:
+                    last_matching_idx = idx
+                if text[last_matching_idx].isalnum() or text[last_matching_idx] == '_':
+                    while (
+                        last_matching_idx + 1 < len(text)
+                        and (text[last_matching_idx + 1].isalnum() or text[last_matching_idx + 1] == '_')
+                    ):
+                        last_matching_idx += 1
+                tokens.append(text[idx : last_matching_idx + 1])
+                idx = last_matching_idx + 1
+                continue
+            if True:
+                # Capture runs of space characters
+                last_space_idx = idx
+                while (
+                    last_space_idx + 1 < len(text)
+                    and text[last_space_idx + 1].isspace()
+                ):
+                    last_space_idx += 1
+                idx = last_space_idx + 1
+        return tokens
+
+    def __init__(self,
+                 add_prefix_space: bool,
+                 version: Tuple[int, int],
+                 bpe_codes: Dict[str, float],
+                 separator: str):
+
+        self.add_prefix_space = add_prefix_space
+        self.version = version
+        self.bpe_codes = bpe_codes
+        self.separator = separator
+
+    def encode(self, text: str) -> List[str]:
+        if self.add_prefix_space:
+            text = f" {text}"
+        return self.helper_encode(text)
+
+    def get_pairs(self, word: List[str]) -> List[Tuple[str, str]]:
+        """
+        Return set of symbol pairs in a word.
+
+        Word is represented as list of symbols (symbols being variable-length strings).
+
+        :param word:
+            word to symbolize
+
+        :return pairs:
+            set of tuples of symbols
+        """
+        pairs: List[Tuple[str, str]] = []
+        prev_char = word[0]
+
+        for char in word[1:]:
+            pairs.append((prev_char, char))
+            prev_char = char
+
+        return pairs
+
+    def bpe(self, orig: str) -> List[str]:
```

Comment:
maybe add a docstring to give the gist of this method?

Reply:
Added